src/qibo/tests/conftest.py

```diff
@@ -11,9 +11,9 @@
 _ACCELERATORS = None
 if ("qibotf" in _available_backends or "qibojit" in _available_backends):
     _ACCELERATORS = "2/GPU:0,1/GPU:0+1/GPU:1,2/GPU:0+1/GPU:1+1/GPU:2"
-if "icarusq" in _available_backends: # pragma: no cover
+if "qiboicarusq" in _available_backends: # pragma: no cover
```

Comment:
We can make this backend name agnostic by using the `.is_hardware` property (which perhaps we need to add to `AbstractBackend`). Alternatively something that could work and skip the tests for all hardware backends would be to change the `_available_backends` definition to:
```Python
_available_backends = set(qibo.K.available_backends.keys()) - set(qibo.K.hardware_backends.keys())
```

In a similar manner, we could possibly extend this PR to completely remove hard-coded backend names by adding the relevant properties in the abstract backend. For example another case which currently requires hard-coded backend names around the code is the multi-GPU (see the `if "qibotf" in _available_backends:` above). We could make this name agnostic using a `.supports_multigpu` property in the backends. The multi-GPU case may be slightly more complicated than `.is_hardware` so we should probably to this in a different PR, but let me know what you think.